api/src/pcapi/core/fraud/api.py

```diff
@@ -238,7 +238,10 @@ def validate_id_piece_number_format_fraud_item(id_piece_number) -> models.FraudI
         return models.FraudItem(
             status=models.FraudStatus.SUSPICIOUS, detail="Le numéro de la pièce d'identité est vide"
         )
-    if not re.fullmatch(r"^[\w]{8,12}|[\s\w]{14}$", id_piece_number):
+    if not re.fullmatch(
+        r"(^\d{18}$)|(^[\w]{8,12}|[\s\w]{14}$)|(^\w{1}\d{6}$)|(^\w{1}\ *\d{8}$)|(^\w{2}\ *\d{7}$)|(^\d{3}\-\d{7}\-\d{2}$)|(^\d{7}$)",
+        id_piece_number,
+    ):
```

Comment:
Je suppose que chaque partie de la regexp correspond à une regexp qui "matche" le format d'un pays. Si c'est le cas, peut-être qu'on pourrait ajouter des commentaires ? Quelque chose comme ça :

```suggestion
    regexp = "|".join((
        r"(^\d{18}$)",  # European Union
        r"(^[\w]{8,12}|[\s\w]{14}$)", # Tunisia
        ...
    ))
    if not re.fullmatch(regexp, id_piece_number):
```


Les mainteneurs de cette regexp apprécieront sûrement. :)

Reply:
ok je fais ça.